File: backend/api/v1/market.py
from fastapi import APIRouter

from backend.schemas.market import HeatmapItem, MarketIndexItem, MarketSentimentResponse
from backend.services.market_service import MarketService

router = APIRouter()
market_service = MarketService()


@router.get("/sentiment", response_model=MarketSentimentResponse)
async def get_market_sentiment():
    return market_service.get_sentiment()


@router.get("/indices", response_model=list[MarketIndexItem])
async def get_market_indices():
    return market_service.get_indices()


@router.get("/heatmap", response_model=list[HeatmapItem])
async def get_market_heatmap():
    return market_service.get_heatmap()

File: frontend/src/api/market.ts
import { apiClient } from './client'
import type { HeatmapItem, MarketIndex, MarketSentiment } from '../types/market'

export async function getMarketSentiment(): Promise<MarketSentiment> {
  const response = await apiClient.get<MarketSentiment>('/market/sentiment')
  return response.data
}

export async function getMarketIndices(): Promise<MarketIndex[]> {
  const response = await apiClient.get<MarketIndex[]>('/market/indices')
  return response.data
}

export async function getMarketHeatmap(): Promise<HeatmapItem[]> {
  const response = await apiClient.get<HeatmapItem[]>('/market/heatmap')
  return response.data
}
